File: back-end/exercise_calendar/src/main/java/com/exercise_calendar/exercise_calendar/dto/CreateReqDto.java
package com.exercise_calendar.exercise_calendar.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CreateReqDto {
    private String category;
    private LocalTime startTime;
    private LocalTime endTime;
    private String description;
    private int rating;
    private LocalDate date;

}

File: front-end/lib/dto/createReqDto.dart
class CreateReqDto {
  final String category;
  final String date;
  final String startTime;
  final String endTime;
  final String description;
  final int rating;

  CreateReqDto({
    required this.category,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.description,
    required this.rating,
  });

  Map<String, dynamic> toJson() => {
        "category": category,
        "date": date,
        "startTime": startTime,
        "endTime": endTime,
        "description": description,
        "rating": rating
      };
}
